piDIOComm.c

```diff
@@ -149,128 +152,97 @@ INT32U piDIOComm_Init(INT8U i8uDevice_p)
 	return 4;		// unknown device
 }
 
-INT32U piDIOComm_sendCyclicTelegram(INT8U i8uDevice_p)
+INT32U piDIOComm_sendCyclicTelegram(u8 devnum)
 {
-	INT32U i32uRv_l = 0;
-	SIOGeneric sRequest_l;
-	SIOGeneric sResponse_l;
-	INT8U len_l, data_out[18], i, p, data_in[70];
-	INT8U i8uAddress;
+	static u8 last_out[40][DIO_OUTPUT_DATA_LEN];
```

Comment:
So this is neither your fault nor actually related to this pull request, but I can't help but notice there's an undocumented limit of 40 DIO modules and there are no checks in place that the limit isn't exceeded. (The kernel will just silently perform out-of-bounds reads and writes.)